src/main/scala/org/broadinstitute/hail/annotations/Annotation.scala

```diff
@@ -57,6 +57,17 @@ object Annotation {
           x.toDouble
       case (JBool(x), TBoolean) => x
 
+      case (JArray(a), TDict(elementType)) =>
```

Comment:
Json objects are string-keyed.  I think you should convert a Dict to/from a JObject.